src/trace.rs

```diff
@@ -271,13 +279,72 @@ impl<S> BroadcastLayer<S> {
 
 impl<S> Layer<S> for BroadcastLayer<S>
 where
-    S: Subscriber + 'static,
+    S: Subscriber + 'static + for<'lookup> LookupSpan<'lookup>,
 {
-    fn on_event(&self, event: &Event<'_>, _ctx: Context<'_, S>) {
-        // Try buffering the event, and if we're not buffering anymore, try to send it along via the
-        // trace sender if it's been established.
-        if !try_buffer_event(event) {
-            try_broadcast_event(event);
+    fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
+        if should_process_tracing_event() {
+            let mut log = LogEvent::from(event);
+            // Add span fields if available
+            if let Some(parent_span) = ctx.event_span(event) {
+                for span in parent_span.scope().from_root() {
+                    if let Some(fields) = span.extensions().get::<SpanFields>() {
+                        for (k, v) in &fields.0 {
+                            log.insert(format!("vector.{}", k).as_str(), v.clone());
```

Comment:
cc/ @fuchsnj could we use the new lookup stuff here to provide a pre-parsed path?

Reply:
I swapped this around to use `path!` in [1556484](https://github.com/vectordotdev/vector/pull/12807/commits/155648489bb47585611e998805842b0361acb581).